server/src/immich/api-v1/asset/asset.service.ts

```diff
@@ -260,7 +260,7 @@ export class AssetService {
 
     try {
       const thumbnailPath = this.getThumbnailPath(asset, query.format);
-      return this.streamFile(thumbnailPath, res, headers);
+      return this.streamFile(thumbnailPath, res, headers, `image/${query.format.toLowerCase()}`);
```

Comment:
We should be a bit more defensive than this. The actual implementation of `thumbnailPath` is questionable.

https://github.com/immich-app/immich/blob/d9e084706f3b7d2f7f38d945b98e6376b2778e95/server/src/immich/api-v1/asset/asset.service.ts#L573

From what I can see, a format of `xyz` will actually return a jpeg, and so serving a jpeg file with `image/xyz` is not great. It also seems that if the webp thumbnail doesn't exist, then the jpeg will be used. Again, serving a jpeg with `image/webp` isn't great. `getThumbnailPath` should probably be changed to not do that.

Reply:
Can we have get thumbnail path return a path _and_ content type, which can be used in stream file? We can hardcore the content type for jpeg or webp depending on which path is used.

Reply:
I changed it so `getThumbnailPath` throws an error if the requested format isn't available (I could also add validation in `getAssetThumbnail` if requested). If there's anything that relies on the current behavior of falling back to JPEG, then @jrasm91's suggestion works too. 

We could possibly combine the two by making `getThumbnailPath`'s `format` parameter optional: fail if an unavailable format is specified, else try to find *something*.

Reply:
It's just tricky changing the behavior of the code, especially when it is so fundamental like serving thumbnails, which is crucial to the main timeline both on the web and mobile app.

I'd probably prefer to see a bug fix to add the content type header without changing any other behavior, personally.

Or, if you do want to change it you need to do some thorough testing of the new behavior to make sure it works on mobile, web, and during a large CLI upload.